feat(typegraphql-crud): added typegraphql-crud example (#2481)
prisma/prisma-examples: .github/readmes/_using-the-graphql-api-typegraphql-crud.md

## Using the GraphQL API

> `typegraphql-prisma` emits the generated TypeGraphql classes to `@generated/typegraphql-prisma` that is located in the `node_modules` whenever `npx prisma generate` is invoked. It also generates a number of model classes, enums and CRUD resolvers and relations resolver based on your `schema.prisma` file. CRUD resolvers support the following operations that are 1:1 matching with the `PrismaClient` API:

Reviewer: ```suggestion
`typegraphql-prisma` emits the generated TypeGraphQL classes to `node_modules/@generated/typegraphql-prisma` whenever `npx prisma generate` is invoked. 

It also generates a number of model classes, enums as well CRUD and relation resolver based on your `schema.prisma` file. The generated CRUD resolvers match the operations of the [Prisma Client API](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference):
```

> - create
> - update
> - delete
> - findUnique
> - findFirst
> - findMany
> - updateMany
> - deleteMany
> - upsert
> - aggregate
> - groupBy

Reviewer: Sorry but I love backticks too much to not use them here 😄 

```suggestion
- `create`
- `update`
- `delete`
- `findUnique`
- `findFirst`
- `findMany`
- `updateMany`
- `deleteMany`
- `upsert`
- `aggregate`
- `groupBy`
```

> Below are example operations that you can send to the API using the GraphQL Playground. You can explore other operations in the Docs section of the GraphQL Playground.

Reviewer: When referring to button names of UI elements, I think it's nice to highlight these in bold. (This creates a visual "anchor" for people to quickly recognize it on a UI as well). For reference: I've used this approach quite a bit in this article where I described a lot of workflows in a browser UI: https://dev.to/prisma/how-to-setup-a-free-postgresql-database-on-heroku-1dc1

```suggestion
Below are example operations that you can send to the API using the GraphQL Playground. You can explore other operations in the **Docs** section of the GraphQL Playground.
```

Feel free to adjust any operation by adding or removing fields. The GraphQL Playground helps you with its auto-completion and query validation features.

#### Retrieve all published posts and their authors

```graphql
query {
  posts {
    id
    title
    content
    published
    author {
      id
      name
      email
    }
  }
}
```

<Details><Summary><strong>See more API operations</strong></Summary>

#### Create a new user

```graphql
mutation {
  createUser(data: {
    name: "Sarah",

>     email: "sarah@prisma.io"

Reviewer: I think `sarah@prisma.io` already exists in the default DB in this case (should this be the case or did you accidentally upload a version of the dev.db that contains this record)? 

This leads to a violation of a uniquye constraint when running the query.

    }
  ) {
    id
  }
}
```

#### Create a new draft

```graphql
mutation {
  createPost(
    data: {
      title: "Join the Prisma Slack",
      content: "https://slack.prisma.io"

>       email: "alice@prisma.io"

Reviewer: The query should be:

```graphql
mutation {
  createPost(
    data: {
      title: "Join the Prisma Slack",
      content: "https://slack.prisma.io"
      author: {
        connect: {
          email: "alice@prisma.io"
        }
      }
    }
  ) {
    id
    published
  }
}
```
